check/base.lua

```diff
@@ -425,23 +425,19 @@ function ChildCheck:_runChild(exePath, exeArgs, environ, callback)
 
   callback = fireOnce(callback)
 
+  local detach_const = process.env['SPAWN_CHECKS_DETACHED'] or false
   self._log(logging.DEBUG, fmt("%s: starting process", exePath))
 
-  local child = childprocess.spawn(exePath, exeArgs, { env = environ })
+  local child = childprocess.spawn(exePath, exeArgs, { env = environ,
+                                                       detached = detach_const})
   child.stdin:destroy() -- close stdin for windows and ruby compatibility
 
   local pluginTimeout = timer.setTimeout(self._timeout, function()
     local timeoutSeconds = (self._timeout / 1000)
 
     self._log(logging.DEBUG, fmt("%s: Plugin didn't finish in %s seconds, killing it...", exePath, timeoutSeconds))
-    --[[ After trying various approaches w.r.t killing a child process within the specified timeout with libuv, none of them seem to work if the childProcess is not detached,
-    since the sleep within 'plugin timeout test' creates a child within a spawned childprocess.
-    A detached process which is created by passing the relevant flag in the options {} to the spawn() command, has its own PGID (process group ID).
-    This makes it easy to kill the entire process group (including its children processes) by prefixing a - (minus sign) to the PGID, example: uv.kill(-child.pid, 'sigkill')
-    We will have to test extensively in staging if we run every plugin with a child process as detached, as the repercussions in production may be unprecedented.
-    ]]
```

Comment:
I wouldn't remove all of these useful comments. They're still very relevant in explaining why we have to worry about spawning detached processes.